cmake build fails in glfw with doxygen installed

Results in:
```
CMake Error at thirdparty/imgui_suite/glfw/CMakeLists.txt:142 (add_subdirectory):
  add_subdirectory given source "docs" which is not an existing directory.
```

Easiest fix is to patch the GLFW thirdparty build to disable the docs build on line 28, or just remove the add_subdirectory.

--- a/thirdparty/imgui_suite/glfw/CMakeLists.txt
+++ b/thirdparty/imgui_suite/glfw/CMakeLists.txt
@@ -25,7 +25,7 @@ endif()
 option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 option(GLFW_BUILD_EXAMPLES "Build the GLFW example programs" ${GLFW_STANDALONE})
 option(GLFW_BUILD_TESTS "Build the GLFW test programs" ${GLFW_STANDALONE})
-option(GLFW_BUILD_DOCS "Build the GLFW documentation" ON)
+option(GLFW_BUILD_DOCS "Build the GLFW documentation" OFF)
 option(GLFW_INSTALL "Generate installation target" ON)
 
 include(GNUInstallDirs)

--- a/upstream_utils/glfw.py
+++ b/upstream_utils/glfw.py
@@ -73,6 +73,7 @@ def main():
 
     patch_list = [
         "0001-Suppress-Compiler-Warnings.patch",
+        "0002-Disable-docs-build-by-default.patch",
     ]
 
     glfw = Lib(name, url, tag, patch_list, copy_upstream_src)

--- a/upstream_utils/glfw_patches/0001-Suppress-Compiler-Warnings.patch
+++ b/upstream_utils/glfw_patches/0001-Suppress-Compiler-Warnings.patch
@@ -1,7 +1,7 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: PJ Reiniger <pj.reiniger@gmail.com>
 Date: Wed, 10 Jul 2024 03:16:58 -0400
-Subject: [PATCH] Suppress Compiler Warnings
+Subject: [PATCH 1/2] Suppress Compiler Warnings
 
 ---
  src/input.c        | 4 ++++

--- a/upstream_utils/glfw_patches/0002-Disable-docs-build-by-default.patch
+++ b/upstream_utils/glfw_patches/0002-Disable-docs-build-by-default.patch
@@ -1,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tyler Veness <calcmogul@gmail.com>
+Date: Sat, 20 Jul 2024 23:18:56 -0700
+Subject: [PATCH 2/2] Disable docs build by default
+
+---
+ CMakeLists.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f5e538bf7cc1807e85a53361ee732122650708b7..cf2dc45788d75adc90c3bc9e49964462740e0723 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -25,7 +25,7 @@ endif()
+ option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
+ option(GLFW_BUILD_EXAMPLES "Build the GLFW example programs" ${GLFW_STANDALONE})
+ option(GLFW_BUILD_TESTS "Build the GLFW test programs" ${GLFW_STANDALONE})
+-option(GLFW_BUILD_DOCS "Build the GLFW documentation" ON)
++option(GLFW_BUILD_DOCS "Build the GLFW documentation" OFF)
+ option(GLFW_INSTALL "Generate installation target" ON)
+ 
+ include(GNUInstallDirs)

